### PR TITLE
User auth

### PR DESCRIPTION
**Pull request by Roystbeef**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -43,6 +47,39 @@ MongoDB.once('open', function() {
     console.log('mongodb connection opened');
 });
 
+// initialize middleware for user auth
+
+app.use(express.static('public'));
+app.use(cookieParser);
+app.use(bodyParser);
+app.use(session({ secret: 'lounge-secret' }));
```
Comment (spencer-brown) [inline app.js]: There are some other settings we want to use here. Check out that screenshot I sent you again, @EvanDWalsh . We need to create a new Redis session object.

File: app.js [hunk 2]
```diff
@@ -43,6 +47,39 @@ MongoDB.once('open', function() {
...
+
+// initialize redis
+var redisUrl;
+var redisPassword;
+if (typeof process.env.REDISTOGO_URL === 'undefined') {
+	redisUrl = url.parse('redis://@127.0.0.1:6379/0');
+	redisPassword = 'evan';
+} else {
+	redisUrl = url.parse(process.env.REDISTOGO_URL);
+	redisPassword = 'NEEDTOSETREDISPASSWORD'
```
Comment (spencer-brown) [inline app.js]: Probably worth leaving a comment here.

File: routes/account.js [hunk 3]
```diff
@@ -7,26 +7,43 @@ module.exports = function(app) {
 
     /* GET sign up. */
     app.get('/account/signup', function(req, res, next) {
+        debugger;
```
Comment (spencer-brown) [inline routes/account.js]: We should remove this debugger.

File: routes/account.js [hunk 4]
```diff
@@ -7,26 +7,43 @@ module.exports = function(app) {
...
         res.render('account/signup');
     });
 
+    app.post('/account/signup', function(req, res, next) {
+
+        debugger;
```
Comment (spencer-brown) [inline routes/account.js]: This one too.

File: routes/account.js [hunk 5]
```diff
@@ -7,26 +7,43 @@ module.exports = function(app) {
...
+        User.register(new User({email : req.body.email}), req.body.password, function(err, user) {
+            if (err) {
+                    console.log(err);
+                    res.redirect('/account/signup');
+            } else {
+                passport.authenticate('local')(req, res, function () {
+                    debugger;
```
Comment (spencer-brown) [inline routes/account.js]: This one too.

**Comment by spencer-brown** [comment]
@Roystbeef can you check this out?

**Comment by Roystbeef** [comment]
LGTM

**Pull request merged**